projects/libs/core-library-typescript/src/common-library/signatures.ts

```diff
@@ -0,0 +1,65 @@
+import * as crypto from "crypto";
+import * as stringify from "json-stable-stringify";
+import * as fs from "fs";
+import * as _ from "lodash";
+
+export interface SignaturesConfig {
+  publicKeysPath?: string;
+  privateKeyPath?: string;
+}
+
+export class Signatures {
+  config: SignaturesConfig;
+  SIGNATURE_FORMAT: crypto.HexBase64Latin1Encoding = "base64";
+  SIGNATURE_TYPE = "sha256";
+
+  private publicKeys = new Map<string, string>();
+  private privateKey: string;
+
+  public constructor(signaturesConfig: SignaturesConfig) {
+    this.config = signaturesConfig;
+
+    this.readPrivateKey();
+    this.readPublicKeys();
+  }
+
+  private readPrivateKey() {
+    if (this.config.privateKeyPath) {
+      this.privateKey = fs.readFileSync(this.config.privateKeyPath).toString();
+    }
+  }
+
+  private readPublicKeys() {
+    if (this.config.publicKeysPath) {
+      fs.readdirSync(this.config.publicKeysPath).forEach((keyName) => {
+        const contents = fs.readFileSync(`${this.config.publicKeysPath}/${keyName}`).toString();
+        this.publicKeys.set(keyName, contents);
+      });
+    }
+  }
+
+  private signObject(object: any, key: string): string {
+    const sign = crypto.createSign(this.SIGNATURE_TYPE);
+    const payload = _.isBuffer(object) ? object : stringify(object);
+    sign.update(payload);
+
+    return sign.sign(key, this.SIGNATURE_FORMAT);
```

Comment:
This enforcement of the encoding (it's not FORMAT btw, it's ENCODING) will break as soon as we support block signatures since we're planning to pass it as a Buffer there, without any encoding

Reply:
Renamed to `SIGNATURE_ENCODING`. We can update it to output buffers when we need it.